packages/calypso-e2e/src/jest-playwright-config/environment.ts

```diff
@@ -171,61 +126,8 @@ class JestEnvironmentPlaywright extends NodeEnvironment {
 			},
 		} );
 
-		// Proxy our E2E browser to bind our custom default context options.
-		// We need to it this way because we're not using the first-party Playwright
-		// and we can't use the playwright.config.js file.
-		// See https://playwright.dev/docs/api/class-testoptions.
-		const wpBrowser = new Proxy( browser, {
-			get: function ( target, prop: keyof Browser ) {
-				const orig = target[ prop ];
-
-				if ( typeof orig !== 'function' ) {
-					return orig;
-				}
-
-				if ( prop === 'newContext' ) {
-					return async function ( options: BrowserContextOptions ): Promise< BrowserContext > {
-						const context = await target.newContext( {
-							...options,
-							...config.contextOptions,
-							recordVideo: { dir: os.tmpdir() },
-						} );
-
-						await context.tracing.start( {
-							screenshots: true,
-							snapshots: true,
-						} );
-
-						return context;
-					};
-				}
-
-				if ( prop === 'newPage' ) {
-					return async function ( options: BrowserContextOptions ): Promise< Page > {
-						const page = await target.newPage( {
-							...options,
-							...config.contextOptions,
-							recordVideo: { dir: os.tmpdir() },
-						} );
-
-						// Set the default timeout for all Playwright methods.
-						// Default value is 15000ms, defined in env-variables.ts.
-						page.setDefaultTimeout( env.TIMEOUT );
-
-						const context = page.context();
-
-						await context.tracing.start( {
-							screenshots: true,
-							snapshots: true,
-						} );
-
-						return page;
-					};
-				}
-
-				return orig.bind( target );
-			},
-		} );
+		// Set up the proxy trap.
+		const wpBrowser = setupBrowserProxyTrap( browser );
```

Comment:
Extracted method is placed below and outside of the `JestEnvironmentPlaywright` class, as the action it performs (intercept and override method calls to the Browser class) isn't tied to the specific instance of the Jest environment.